include/fmt/format.h

```diff
@@ -2435,8 +2435,7 @@ FMT_CONSTEXPR20 auto do_write_float(OutputIt out, const DecimalFP& f,
   auto significand = f.significand;
   int significand_size = get_significand_size(f);
   const Char zero = static_cast<Char>('0');
-  auto sign = fspecs.sign;
```

Comment:
Please keep `sign` definition here, we don't want to capture all specs. Perhaps changing `auto` to `sign_t` will workaround the intel compiler bug?

Reply:
sign_t doesn’t work too, the alternative is to replace the lambda capture by value, but maybe we can cancel this PR and just wait, KornevNikita said they fixed it already, just need to upstream to the SYCL branch